### PR TITLE
Support for searching multiple races in characters

### PR DESCRIPTION
**Pull request by spitfire305**

### CODE REVIEW

File: app/Models/Concerns/Filterable.php [hunk 1]
```diff
@@ -315,7 +343,7 @@ protected function extractSearchOperator($value, string $key): array
     {
         $operator = 'like';
         $filterValue = $value;
-        if ($key !== 'tags') {
+        if (($key !== 'tags') && ($key !== 'races')) {
```
Comment (ilestis) [inline app/Models/Concerns/Filterable.php:346]: Is this change actually needed?

**Review by ilestis — CHANGES_REQUESTED** [review]
One tweak for code readibility, and another to confirm a change is needed

File: app/Models/Concerns/Filterable.php (hunk 1 above)
Comment (spitfire305) [reply app/Models/Concerns/Filterable.php:346]: Yes, if that change is not made the following error message is displayed when trying to search using races as a parameter:

`strncmp(): Argument #1 ($string1) must be of type string, array given`

File: app/Models/Concerns/Filterable.php [hunk 2]
```diff
@@ -125,6 +125,34 @@ public function scopeFilter(Builder $query, $params)
...
+                        }
+                        $query = $this->joinEntity($query);
+
+                        // Make sure we always have an array
+                        if (!is_array($value)) {
+                            $value = [$value];
+                        }
+
+                        if (!empty($filterOption) && $filterOption == 'exclude') {
+                            $tagIds = [];
```
[the remark this answers is not in the document]
Comment (spitfire305) [reply app/Models/Concerns/Filterable.php:141]: It seems I forgot to rename the variables, sorry!
